### PR TITLE
Implemented mongo db connection and repository

### PR DESCRIPTION
**Pull request by mmackiewicz1**
Added configuration and repository. As requested changed "id" in EhrExtractStatus collection to "extractId".

### CODE REVIEW

File: service/build.gradle [hunk 1]
```diff
@@ -26,10 +26,14 @@ repositories {
...
 	implementation 'org.springframework.boot:spring-boot-starter-actuator'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
 	implementation 'ch.qos.logback:logback-classic:1.2.3'
 	implementation 'ch.qos.logback.contrib:logback-json-classic:0.1.5'
 	implementation 'ch.qos.logback.contrib:logback-jackson:0.1.5'
-	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'org.apache.commons:commons-lang3:3.11'
+
+
```
Comment (Broom94) [inline service/build.gradle:36]: Do we need whitespace here

Comment (mmackiewicz1) [reply service/build.gradle:36]: I will remove it.

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -9,3 +9,14 @@ management:
...
+  uri: ${GP2GP_MONGO_URI:mongodb://localhost:27017}
+  database: ${GP2GP_MONGO_DATABASE_NAME:gp2gp}
+  host: ${GP2GP_MONGO_HOST:}
+  port: ${GP2GP_MONGO_PORT:}
+  username: ${GP2GP_MONGO_USERNAME:}
+  password: ${GP2GP_MONGO_PASSWORD:}
+  options: ${GP2GP_MONGO_OPTIONS:}
+  autoIndexCreation: ${GP2GP_MONGO_AUTO_INDEX_CREATION:true}
+  ttl: ${GP2GP_MONGO_TTL:P30D}
+  cosmosDbEnabled: ${GP2GP_COSMOS_DB_ENABLED:false}
```
Comment (Broom94) [inline service/src/main/resources/application.yml:22]: Please add environment variables to ./README.md +  update ./docker/docker-compose.yml with these new variables

**Review by Broom94 — CHANGES_REQUESTED** [review]
Good job on this one, those log message are really clear and easy to understand! Should we have a test container setup for the db?

File: README.md [hunk 3]
```diff
@@ -21,9 +21,18 @@ Variables without a default value and not marked optional, *MUST* be defined for
...
+| GP2GP_MONGO_HOST                   | (*)                       | Mongo database host. Can be left blank if full connection string is provided.
+| GP2GP_MONGO_USERNAME               | (*)                       | Mongo database username. Can be left blank if full connection string is provided.
+| GP2GP_MONGO_PASSWORD               | (*)                       | Mongo database password. Can be left blank if full connection string is provided.
+| GP2GP_MONGO_OPTIONS                | (*)                       | Mongodb URL encoded parameters for the connection string without a leading "?". Can be left blank if full connection string is provided.
+| GP2GP_MONGO_AUTO_INDEX_CREATION    | true                      | (Optional) Should auto index for Mongo database be created.
+| GP2GP_MONGO_TTL                    | P30D                      | (Optional) Time-to-live value for inbound and outbound state collection documents as an [ISO 8601 Duration](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+| GP2GP_COSMOS_DB_ENABLED            | false                     | (Optional) If true the adaptor will enable features and workarounds to support Azure Cosmos DB.
```
Comment (Broom94) [inline README.md:36]: Please add necessary variables added in the environment variable list to the section below 'How to run service'

File: service/src/main/java/uk/nhs/adaptors/gp2gp/TestClass.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package uk.nhs.adaptors.gp2gp;
+
+public class TestClass {
+}
```
Comment (Broom94) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/TestClass.java:4]: Remove?

**Review by Broom94 — CHANGES_REQUESTED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/TestClass.java (hunk 4 above)
Comment (mmackiewicz1) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/TestClass.java:4]: Sorry, commited it by accident.

**Review by mattd-kainos — APPROVED** [review]

**Review by Broom94 — APPROVED** [review]

**Pull request merged**